src/systems/filecoin_markets/deal/_index.md

```diff
@@ -65,3 +65,34 @@ On every PoSt Submission, the following steps happen.
 # Deal Flow
 
 {{< diagram src="diagrams/deal-flow.mmd.svg" title="Deal Flow Sequence Diagram" >}}
+
+# Deal States
+
+All on-chain economic activities in Filecoin start with the deal. This section aims to explain different states of a deal and their relationship with other concepts in the protocol such as Power, Payment, and Collaterals.
+
+A deal has the following states:
+
+- `Unpublished`: the deal has yet to be posted on chain.
+- `Published`: the deal has been published and accepted by the chain but is not yet active as the sector containing the deal has not been proven.
+- `Active`: the deal has been proven and not yet expired.
+- `Deleted`: the deal has expired or the sector containing the deal has been terminated because of faults.
+
+Note that `Unpublished` and `Deleted` states not tracked on chain. To reduce on-chain footprint, an `OnchainDeal` struct is created when a deal is published and it keeps track of a `LastPaymentEpoch` which defaults to zero when a deal is in the `Published` state. A deal transitions into the `Active` state when `LastPaymentEpoch` is positive.
+
+The following describes how a deal transitions between its different states.
+
+- `Unpublished -> Published`: this is triggered by `StorageMarketActor.PublishStorageDeals` which validates new storage deals, locks necessary funds, generates deal IDs, and registers the storage deals in `StorageMarketActor`.
+- `Published -> Deleted`: this is triggered by `StorageMinerActor.ProveCommitSector` during InteractivePoRep when the elapsed number of epochs between PreCommit and ProveCommit messages exceeds `MAX_PROVE_COMMIT_SECTOR_EPOCH`. ProveCommitSector will also trigger garbage collection on the list of published storage deals.
+- `Published -> Active`: this is triggered by `ActivateStorageDeals` after successful `StorageMinerActor.ProveCommitSector`. It is okay for the StorageDeal to have already started (i.e. for `StartEpoch` to have passed) at this point but it must not have expired.
```

Comment:
What happens if it has expired? Is that another case of Published -> Deleted?

Reply:
Yes. Should maybe be explicited @zx